### DISCUSSION TITLE
Cannot Add SharePoint item to list (Error 401)

### DISCUSSION THREAD

**Discussion by dietcheesecake** [pnp/pnpcore#825] [Q&A]
Hi,

I have an issue that I cannot add item to SharePoint list. Can anyone help me on this issue? Please see my code below...

```
var host = Host.CreateDefaultBuilder().ConfigureServices((hostingContext, services) =>
{
    // Configure PnP services
    services.AddPnPCore(options =>
    {
        options.DefaultAuthenticationProvider = new InteractiveAuthenticationProvider();
    });
}).UseConsoleLifetime().Build();

await host.StartAsync();

var pnpContextFactory = host.Services.GetRequiredService<IPnPContextFactory>();

using (var context = await pnpContextFactory.CreateAsync(new Uri(url)))
{
    await context.Web.LoadAsync(p => p.Title);
    Console.WriteLine($"The title of the web is {context.Web.Title}");

    var pnpList = context.Web.Lists.GetByTitle("PnP", p => p.Title);

    var addedItem = await pnpList.Items.AddAsync(new Dictionary<string, object>()
    {
        { "Title", "PnP1" }
    });
}
```

I get an Exception
Message: SharePoint Rest service exception.
Error.Message: 401 UNAUTHORIZED

I tried with all add methods (pnpList.Items.AddAsync, pnpList.Items.AddBatchAsync, pnpListItems.Add). They all give the same error.

For more information
** I can create list and update list item, but NOT add new item.
** I login as Site Collection Administrator.

Thank you,
Chris L,

**Comment by jansenbe** [comment]
@dietcheesecake : what permissions did you grant to the Azure AD app you're using? If you've not specifying an app in your setup, you're defaulted to the PnP Management Shell app. That one however needs to be consented in your Azure AD environment before you can use it. See https://pnp.github.io/pnpcore/using-the-sdk/configuring%20authentication.html for our authentication documentation. See https://pnp.github.io/powershell/articles/authentication.html#setting-up-access for using the PnP PowerShell management app.
